sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AvoidExcessiveInheritance.cs

```diff
@@ -103,6 +106,15 @@ protected override void Initialize(ParameterLoadingAnalysisContext context)
                 }, SyntaxKind.ClassDeclaration);
         }
 
+        private static string GetRootNamespace(INamedTypeSymbol namedSymbol)
+        {
+            return namedSymbol
+                        .ContainingNamespace
+                        .ToDisplayString()
+                        .SplitCamelCaseToWords()

```

Comment:
I think you can use a split on . (dot) and take first instead of using this heavy `SplitCamelCaseToWords`.